xla/service/cpu/onednn_matmul_rewriter.cc

```diff
@@ -467,7 +467,7 @@ class OneDnnMatMulRewriteVisitor : public DfsHloRewriteVisitor {
         OptionalConvertAndBitcast(&optional_dot_convert, &optional_dot_bitcast,
                                   OneDnnMatmulInstr(&dot))
             .WithOneUser(),
-        m::Op(&addend_intermediate).WithOneUser());
```

Comment:
Can you please extend the PR description to explain why we had this restriction before and why we're removing it now?

Reply:
The assumption was that addend can have only 1 user but it can have multiple users as seen in some models. So, removing this restriction enables the fusion for those cases.